### PR TITLE
Issue 73

### PR DESCRIPTION
**Pull request by ViRu-ThE-ViRuS**
#73 

### changes made
- cannot `Join a project` unless user agreement is signed
- cannot add users to a project who have not yet signed user agreement

### improvements
- update the message displayed when 0 users are selected to add to project

### CODE REVIEW

File: coldfront/core/project/templates/project/project_list.html [hunk 1]
```diff
@@ -10,10 +10,18 @@
...
-    </a>
+
+    {% if user_agreement_signed %}
+      <a class="btn btn-primary" href="{% url 'project-join-list' %}" role="button">
+        <i class="fas fa-user-plus" aria-hidden="true"></i>
+        Join a project
+      </a>
+    {% else %}
+      <button class="btn btn-primary" disabled="true">
+        Join a project
```
Comment (matthew-li) [inline coldfront/core/project/templates/project/project_list.html:21]: Minor styling issue: let's retain the icon from the active case.

File: coldfront/core/project/views.py [hunk 2]
```diff
@@ -1464,10 +1468,20 @@ def post(self, request, *args, **kwargs):
         return redirect(next_view)
 
 
-class ProjectJoinListView(ProjectListView):
+class ProjectJoinListView(ProjectListView, UserPassesTestMixin):
```
Comment (matthew-li) [inline coldfront/core/project/views.py:1471]: This looks good. We should also block `ProjectJoinView` similarly.

**Review by matthew-li — CHANGES_REQUESTED** [review]
Thanks for working on this--the changes look good! I think there's one case not being handled, which I mentioned in the comments.

Also, @kmuriki, should we prevent project managers from adding a user to their project (via the "Add Users" button) until that user has signed the agreement? Maybe we can display the user as grayed out, and write some explanatory text at the top saying that grayed out users need to sign the agreement first. What do you think?

**Comment by kmuriki** [comment]
Yes greying out unsigned users for adding to a project to the PI sounds great!

**Comment by ViRu-ThE-ViRuS** [comment]
Ready for review!

**Review by matthew-li — CHANGES_REQUESTED** [review]
Thanks for adding the blocks. Can you also add the following explanation somewhere on the "Add Users" view?

> You may only add users who have signed the access agreement.

**Review by matthew-li — APPROVED** [review]
I've added the text, since it was a quick addition, and resolved some merge conflicts. Thanks again!

**Pull request merged**